### PR TITLE
UserProjectViewModel

### PR DESCRIPTION
**Pull request by ILMTitan**
Extract GcpUserProjectViewModel from CloudExploererViewModel.

This PR depends upon PRs #1037, #1038 and #1039.

Commits for this PR start at 00f6683.

This PR is part of PR #1035

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/CloudExplorerViewModel.cs [hunk 1]
```diff
@@ -166,21 +128,16 @@ public ICommand EmptyStateCommand
...
 
-        private static IGPlusDataSource GPlusDataSource => DataSourceFactory.Default.GPlusDataSource;
-
-        private static IResourceManagerDataSource ResourceManagerDataSource =>
-            DataSourceFactory.Default.ResourceManagerDataSource;
-
         #region ICloudSourceContext implementation.
 
-        Project ICloudSourceContext.CurrentProject => _currentProject;
+        Project ICloudSourceContext.CurrentProject => UserProject.CurrentProjectAsync.Value;
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/CloudExplorerViewModel.cs]: Should this be `UserProject.CurrentProjectAsync?.Value`?

File: GoogleCloudExtension/GoogleCloudExtension/MenuBarControls/IGcpUserProjectViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        /// The command to execute to select a new GCP project.
+        /// </summary>
+        IProtectedCommand SelectProjectCommand { get; }
+
+        /// <summary>
+        /// The command to open the GCP Menu Bar Popup.
+        /// </summary>
+        ICommand OpenPopup { get; }
+
+        void LoadCurrentProject();
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension/MenuBarControls/IGcpUserProjectViewModel.cs]: I think we should add some comments for these functions and their relationships with the other `AsyncProperty`?

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/CloudExplorerViewModel.cs (hunk 1 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension/CloudExplorer/CloudExplorerViewModel.cs]: No. I set `CurrentProjectAsync` in the constructor of the class, and never assign it to null.

File: GoogleCloudExtension/GoogleCloudExtension/MenuBarControls/IGcpUserProjectViewModel.cs (hunk 2 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension/MenuBarControls/IGcpUserProjectViewModel.cs]: Done.

**Comment by ILMTitan** [comment]
PTAL

**Pull request merged**